dCommon/GeneralUtils.cpp

```diff
@@ -202,18 +202,18 @@ bool GeneralUtils::CaseInsensitiveStringCompare(const std::string& a, const std:
 // MARK: Bits
 
 //! Sets a specific bit in a signed 64-bit integer
-int64_t GeneralUtils::SetBit(int64_t value, uint32_t index) {
-	return value |= 1ULL << index;
+int64_t GeneralUtils::SetBit(int64_t value, eObjectBits index) {
+	return value |= 1ULL << static_cast<uint32_t>(index);
 }
 
 //! Clears a specific bit in a signed 64-bit integer
-int64_t GeneralUtils::ClearBit(int64_t value, uint32_t index) {
-	return value &= ~(1ULL << index);
+int64_t GeneralUtils::ClearBit(int64_t value, eObjectBits index) {
+	return value &= ~(1ULL << static_cast<uint32_t>(index));
 }
 
 //! Checks a specific bit in a signed 64-bit integer
-bool GeneralUtils::CheckBit(int64_t value, uint32_t index) {
-	return value & (1ULL << index);
+bool GeneralUtils::CheckBit(int64_t value, eObjectBits index) {
+	return value & (1ULL << static_cast<uint32_t>(index));
 }
```

Comment:
Real question: could I avoid this static cast with an operator overload

Reply:
I may have an idea to alleviate these